ddtrace/opentelemetry/span.go

```diff
@@ -25,7 +25,12 @@ var _ oteltrace.Span = (*span)(nil)
 
 type span struct {
 	tracer.Span
-	finished   bool
+	finished bool
+	// nameSet signifies if the span operation name was set with specifically `operation.name` tag
+	// it is persisted here since it's not possible to see how the name was set,
+	// nor access the span attributes set before
+	nameSet    bool
```

Comment:
After playing around with it for a bit, I don't see a way around it without preserving the state of the attributes. 
See this [PR](https://github.com/DataDog/system-tests/pull/1819). For example, if the attributes that are needed in a pair, like 'messaging.system' or 'messaging.operation' are set separately, it will mess up with the entire calculation. But even if we decide to disregard this corner case, still we need to know if the span name was set explicitly or not.

Reply:
After syncing offline, we decided to go forward with persisting the attributes in the local span struct instead.